lucene/core/src/test/org/apache/lucene/util/bkd/TestBKD.java

```diff
@@ -1063,7 +1064,7 @@ public void testWastedLeadingBytes() throws Exception {
 
     Directory dir = newFSDirectory(createTempDir());
     int numDocs = 100000;
-    BKDWriter w = new BKDWriter(numDocs+1, dir, "tmp", numDims, bytesPerDim, 32, 1f, numDocs, true);
+    BKDWriter w = new BKDWriter(numDocs+1, dir, "tmp", numDims, numDims, bytesPerDim, 32, 1f, numDocs, true);
```

Comment:
Couldn't we add a mode on the random tests where dataDim != indexedDim. The test can only check that the data dims are properly propagated to the leaf nodes.

Reply:
I have just seen there is already a test using different index and data dimensions, still it would be worthy to check if possible that data dimensions are properly propagated.

Reply:
Thanks @nknize! 
Would it be possible to add the following assertion in the method for IntersectVisitor#visit. That would signal that we are propagating the index and data dimensions to the method?

```
   byte[][] expectedDocs = docValues[docID];
   for (int dim =0; dim < numDataDims; dim++) {
        assertTrue(FutureArrays.compareUnsigned(packedValue, dim * numBytesPerDim, dim * numBytesPerDim + numBytesPerDim, expectedDocs[dim], 0, numBytesPerDim) == 0);
    }
```

Reply:
I have realised that the above assertion is incorrect for multivalued docs so please discard it. Maybe we can just assert the number of bytes in packedValue.